pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/RecordLocation.java

```diff
@@ -24,6 +24,7 @@
 /**
  * Indicate a record's location on the local host.
  */
+@SuppressWarnings("rawtypes")
 public class RecordLocation {
   private final IndexSegment _segment;
```

Comment:
for anything other than Hashmap, we can't store Segment reference in the store. Should we replace it with the `int` or something else. We might also need to add method in the interface to convert the integer segmentId to segment object.

Reply:
Moved it into the implementation as a helper class